app/pulp/app/models/progress.py

```diff
@@ -70,14 +77,24 @@ def save(self, *args, **kwargs):
         """
         if self.task_id is None:
             self.task_id = Task.objects.get(id=util.get_current_task_id())
-        super(ProgressReport, self).save(*args, **kwargs)
+
+        if self._using_context_manager:
+            now = timezone.now()
+            if now - self._last_increment_time >= datetime.timedelta(milliseconds=_batch_interval):
+                self._last_increment_time = now
+                super(ProgressReport, self).save(*args, **kwargs)
+        else:
+            super(ProgressReport, self).save(*args, **kwargs)
 
     def __enter__(self):
         """
         Saves the progress report state as RUNNING
         """
         self.state = self.RUNNING
+        self._using_context_manager = True
```

Comment:
I think this will cause the transition to Running to occur 0.5 seconds after it really does. I think we want the first save() to occur immediately. You could fix this by calling save() before setting `self._using_context_manager = True`.